File: typescript/1426-counting-elements.ts
/*
    1426 - Counting Elements
    Hashset | Check existence
    Time: O(n) | Space: O(n)
*/

function countElements(arr: number[]): number {
  let res = 0;
  const set: Set<number> = new Set(arr);

  for (const i of arr) {
    if (set.has(i + 1)) {
      res++;
    }
  }

  return res;
}

File: python/1426-counting-elements.py
"""
1426 - Counting Elements [Easy]
Hash Set | Check Existence | Array
Time: O(n) | Space: O(n)

Given an integer array arr, count how many elements x there are, such that x + 1 is also in arr. If there are duplicates in arr, count them separately.
"""

from typing import List


class Solution:
    def countElements(self, arr: List[int]) -> int:
        res = 0
        num_set = set(arr)

        for i in arr:
            if i + 1 in num_set:
                res += 1

        return res
